### PR TITLE
First PR, ISMB abstract results

### PR DESCRIPTION
**Pull request by nrosed**
This is a new application of BuDDI, where we use ATAC-seq and cfDNA data instead of sc/bulk RNA-Seq. The set-up is mostly the same, except instead of genes we will use regions. The main goal of this analysis is to see if I can predict the tumor type in a sample, where it is present at different abundances. To simulate this, we add PBMC cells at different amounts to a bulk TCGA sample.

Here is an overview of how this is done in this repo:

- `preprocessing_qc/process_cfdna.sh` and `preprocessing_qc/run_process_cfdna.sh`
    - shell script to format the cfDNA files
- `preprocessing_qc/process_cfdna.ipynb`
    - process the cfDNA samples into a format that contains the regions of interest
- `preprocessing_qc/process_pbmc.ipynb`
    - QC / reformat the PBMC data to get them ready to make pseudobulks.
- `preprocessing_qc/process_tcga.ipynb`
    - QC / reformat the TCGA data to get them ready to make pseudobulks.
- `preprocessing_qc/make_pseudobulk.ipynb`
    - Make pseudobulks with different numbers of PBMCs and a single TCGA sample
    - we make sure we have a full heldout set for pseudobulks with PBMCs and TCGA samples that have never been used in any other pseudobulk
-  `tcga_analysis/tcga_hyp_search.ipynb`
    - simple analysis, using only the ATAC data to predict the TCGA sample from the pseudobulks
    - we see if we can predict cfDNA origin using this simple model
- `tcga_analysis/tcga_cfdna.ipynb`
    - since the cfDNA samples were soo different than the pseudobulks, we need to integrate a subset of them in the training procedure 
    - we find that this is still not enough for predicting the label of unseen cfDNA samples. Instead, I will extend BuDDI such that we can train with the proportions and the cancer label being semi-supervised
- `tcga_analysis/tcga_comparators.ipynb`
   - using ATAC data alone, we see if other methods can also predict the TCGA sample. We compare with RBF-SVM, AdaBoost, and MLP. BuDDI out performs in the higher blood proportion examples.

### CODE REVIEW

File: .gitignore
Comment (ivichadriana) [inline .gitignore:95]: do you want these explanation in the .gitignore?

File: preprocessing_qc/run_process_cfdna.sh
Comment (ivichadriana) [inline preprocessing_qc/run_process_cfdna.sh:8]: you could iterate through the in_names and ct_names so it looks more stylish

**Comment by ivichadriana** [comment]
adding RviewNB comments now

File: preprocessing_qc/make_pseudobulk.ipynb
Comment (ivichadriana) [inline preprocessing_qc/make_pseudobulk.ipynb:1]: <p>can you write where the data comes from a little? we only have tcga and pbmc as info</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-1579763921'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

Comment (ivichadriana) [inline preprocessing_qc/make_pseudobulk.ipynb:1]: <p>empty spaces</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-1579763922'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='4'/>

Comment (ivichadriana) [inline preprocessing_qc/make_pseudobulk.ipynb:1]: <p>"that we are on expected scaling"?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-1579763927'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='5'/>

Comment (ivichadriana) [inline preprocessing_qc/make_pseudobulk.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #23.    num_cells_vec = [19000, 99000, 99900]  # 5% 1% .1%
</pre><p>you assign this num_cells_vec twice:</p><p># simulate different number of cells</p><p>num_cells_vec = [100, 500, 1000, 5000, 5001]&nbsp;# 5001 indicates that it should be ALL blood, using 5000 cells</p><p>num_cells_vec = [19000, 99000, 99900]&nbsp;# 5% 1% .1%</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-1579763931'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='21'/>

Comment (ivichadriana) [inline preprocessing_qc/make_pseudobulk.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #40.        #tcga_adata_10 = curr_tcga_adata.concatenate(curr_tcga_adata,
</pre><p>commented out section</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-1579763935'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='21'/>

File: preprocessing_qc/process_cfdna.ipynb
Comment (ivichadriana) [inline preprocessing_qc/process_cfdna.ipynb:1]: <p>you could add more description of what the notebook does and what the data is </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-1579763976'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

Comment (ivichadriana) [inline preprocessing_qc/process_cfdna.ipynb:1]: <p>this funct could go in other file with other functions</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-1579763979'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='3'/>

Comment (ivichadriana) [inline preprocessing_qc/process_cfdna.ipynb:1]: <p>""" commented out for now -- file too big</p><p> </p><p>just making sure you want to push with these commented out</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-1579763984'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='8'/>

File: preprocessing_qc/process_tcga.ipynb
Comment (ivichadriana) [inline preprocessing_qc/process_tcga.ipynb:1]: <p>you could add a bit more description of what are we looking at/where data comes from and is processed</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-1579764096'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

**Comment by ivichadriana** [comment]
i think because i clicked "Comment" now i can't approve without you doing something (approve button is gone?) anyways, some suggestions here and there, nothing major. Mainly the notebook descriptions. i'll see if i can approve after you open the comments.

**Review by ivichadriana — APPROVED** [review]
Ah, I figured it out. Approved.

File: .gitignore
Comment (nrosed) [reply .gitignore:95]: yes, I wanna keep this

File: preprocessing_qc/run_process_cfdna.sh
Comment (nrosed) [reply preprocessing_qc/run_process_cfdna.sh:8]: I think for 2 its ok -- but I will keep this in mind for larger argument lists

File: preprocessing_qc/make_pseudobulk.ipynb
Comment (nrosed) [reply preprocessing_qc/make_pseudobulk.ipynb:1]: <p>added more details</p>

Comment (nrosed) [reply preprocessing_qc/make_pseudobulk.ipynb:1]: <p>deleted</p>

Comment (nrosed) [reply preprocessing_qc/make_pseudobulk.ipynb:1]: <p>added clarification: "Since we are combining single cell and atac, we need to make sure that the bulk and pseudobulk values are on the same scale; meaning one dataset isn't 10 times larger than the other."</p>

Comment (nrosed) [reply preprocessing_qc/make_pseudobulk.ipynb:1]: <p>concatenated the  vector together. I did them separately before to speed it up, but I don't need to do that any more</p>

Comment (nrosed) [reply preprocessing_qc/make_pseudobulk.ipynb:1]: <p>removed -- this was back when the data frame was too big to load into memory, and I had to subdivide it; I don't need to do that anymore.</p>

File: preprocessing_qc/process_cfdna.ipynb
Comment (nrosed) [reply preprocessing_qc/process_cfdna.ipynb:1]: <p>Yes, this is a good point. I am going to change the folder structure of this project once I get more cfDNA data. When I do that I will make a preprocessing methods file to store all the methods.</p>

Comment (nrosed) [reply preprocessing_qc/process_cfdna.ipynb:1]: <p>yes -- keeping them for reference later when I am able to do more specific region filtering / score calculation.</p>

Comment (nrosed) [reply preprocessing_qc/process_cfdna.ipynb:1]: <p>added in <a href="https://github.com/greenelab/buddi_atac/pull/1/commits/f75d90dc63400fc3a0e34ee2ffb8df26a5a35608" target="_blank">f75d90d</a></p>

File: preprocessing_qc/process_tcga.ipynb
Comment (nrosed) [reply preprocessing_qc/process_tcga.ipynb:1]: <p>added in <a href="https://github.com/greenelab/buddi_atac/pull/1/commits/4cfd7789337a8585751a287d95bc47e48139594a" target="_blank">4cfd778</a></p>

**Comment by nrosed** [comment]
<p>added in <a href="https://github.com/greenelab/buddi_atac/pull/1/commits/75c59a9812d7ddc01d22e2e3d69a47822e3775a6" target="_blank">75c59a9</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-2077656154'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>deleted in </p><p> </p><p><code><a href="https://github.com/greenelab/buddi_atac/pull/1/commits/f87c6e420526598299af9702eaf8ba03555dbcbf" target="_blank">f87c6e4</a></code></p><p> </p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-2077656187'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>updated description in </p><p> </p><p><code><a href="https://github.com/greenelab/buddi_atac/pull/1/commits/f87c6e420526598299af9702eaf8ba03555dbcbf" target="_blank">f87c6e4</a></code></p><p> </p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-2077656208'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>fixed in </p><p> </p><p><code><a href="https://github.com/greenelab/buddi_atac/pull/1/commits/633b4e1a43afafed12447d6ac77c03164cb5ad2c" target="_blank">633b4e1</a></code></p><p> </p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-2077656493'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>fixed in </p><p> </p><p><code><a href="https://github.com/greenelab/buddi_atac/pull/1/commits/f87c6e420526598299af9702eaf8ba03555dbcbf" target="_blank">f87c6e4</a></code></p><p> </p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-2077656303'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>moved in </p><p> </p><p><code><a href="https://github.com/greenelab/buddi_atac/pull/1/commits/f87c6e420526598299af9702eaf8ba03555dbcbf" target="_blank">f87c6e4</a></code></p><p> </p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-2077656262'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>removed in </p><p> </p><p><code><a href="https://github.com/greenelab/buddi_atac/pull/1/commits/f87c6e420526598299af9702eaf8ba03555dbcbf" target="_blank">f87c6e4</a></code></p><p> </p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-2077656275'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>atac seq is pretty cool. I'm starting to think that it's better to call cell-types from ATAC-Seq than expression, because the signals are closer to binary on-off signals. not sure yet, though.</p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-2077656095'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>removed in </p><p> </p><p><code><a href="https://github.com/greenelab/buddi_atac/pull/1/commits/e207f320eb090fe814ead1555ec18cedf734589b" target="_blank">e207f32</a></code></p><p> </p><p> </p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-2077656072'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>removed in </p><p> </p><p><code><a href="https://github.com/greenelab/buddi_atac/pull/1/commits/e207f320eb090fe814ead1555ec18cedf734589b" target="_blank">e207f32</a></code></p><p> </p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-2077656046'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>moved in </p><p> </p><p><code><a href="https://github.com/greenelab/buddi_atac/pull/1/commits/1a05c7995755cacf649f07b98846f80b52603135" target="_blank">1a05c79</a></code></p><p> </p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-2077656541'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>fixed in </p><p> </p><p><code><a href="https://github.com/greenelab/buddi_atac/pull/1/commits/1a05c7995755cacf649f07b98846f80b52603135" target="_blank">1a05c79</a></code></p><p> </p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-2077656520'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>added in </p><p> </p><p><code><a href="https://github.com/greenelab/buddi_atac/pull/1/commits/1a05c7995755cacf649f07b98846f80b52603135" target="_blank">1a05c79</a></code></p><p> </p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-2077656471'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>i deleted this whole section because its actually not needed in this notebook</p><p> </p><p> </p><p><code><a href="https://github.com/greenelab/buddi_atac/pull/1/commits/1a05c7995755cacf649f07b98846f80b52603135" target="_blank">1a05c79</a></code></p><p> </p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-2077656389'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by nrosed** [comment]
<p>added in </p><p> </p><p><code><a href="https://github.com/greenelab/buddi_atac/pull/1/commits/1a05c7995755cacf649f07b98846f80b52603135" target="_blank">1a05c79</a></code></p><p> </p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_atac/pull/1/discussion/#comment-2077656349'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by ivichadriana** [comment]
Do you need my review? got a notification but it says merged

**Comment by nrosed** [comment]
You reviewed it in April, I just finally got around to addressing them!

Sorry for the delay!

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Pull request merged**
